### PR TITLE
Adding checkpointing for .zip file parsing

### PR DESCRIPTION
**Pull request by KPHippe**
# Description
Adding support for resume from existing output directory for .zip inputs. 
Adding small CLI function to zip PDF's together

Note: The method used for looking for parsed inputs will only work on.zip inputs. This is 
a by product of the way logging is handled for .pdfs (the batches do not log each individual batch being processed in the same way the parsing a zip file logs the root zip file that contains the whole batch being processed). If we need to add support for resuming from a parsing run that uses .pdf files, we need to explicitly log these in a different manner. 

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (internal implementation changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (no changes to the code)
- [ ] CI change (changes to CI workflows, packages, templates, etc.)
- [ ] Version changes (changes to the package or dependency versions)

## Testing
Testing on small zip input on polaris

N/A

## Pull Request Checklist

Please confirm the PR meets the following requirements.
- [ ] Relevant tags are added (breaking, bug, dependencies, documentation, enhancement, refactor).
- [X] Code changes pass `pre-commit` (e.g., ruff, mypy, etc.).
- [ ] Tests have been added to show the fix is effective or that the new feature works.
- [ ] New and existing unit tests pass locally with the changes.
- [ ] Docs have been updated and reviewed if relevant.

### CODE REVIEW

**Comment by KPHippe** [comment]
Time logger now used to parse the logs for checkpoint information

File: pdfwf/cli.py [hunk 1]
```diff
@@ -434,6 +435,71 @@ def parse_timers(
...
+    from concurrent.futures import as_completed
+    from concurrent.futures import ProcessPoolExecutor
+
+    from pdfwf.utils import zip_worker
+
+    # Make output directory if it does not already exist
+    output_dir.mkdir(exist_ok=True, parents=True)
+
+    # Get all PDF files in the directory
+    pdf_files = list(root_dir.glob(glob_pattern))[:1000]
```
Comment (braceal) [inline pdfwf/cli.py]: Should probably remove the 1000 file cutoff?

File: pdfwf/cli.py [hunk 2]
```diff
@@ -434,6 +435,71 @@ def parse_timers(
...
+    total_files = len(pdf_files)
+    print(f'Found {total_files} PDF files.')
+
+    # Calculate the number of chunks
+    num_chunks = (total_files + chunk_size - 1) // chunk_size
+
+    with ProcessPoolExecutor(max_workers=num_cpus) as pool:
+        futures = []
+        for i in range(num_chunks):
```
Comment (braceal) [inline pdfwf/cli.py]: There is a utils function `batch_data` that might help here.

File: pdfwf/cli.py (hunk 1 above)
Comment (KPHippe) [reply pdfwf/cli.py]: lol yes oops

File: pdfwf/cli.py (hunk 2 above)
Comment (KPHippe) [reply pdfwf/cli.py]: great

**Pull request merged**